pkg/alertmanager/multitenant.go

```diff
@@ -429,7 +431,7 @@ func (am *MultitenantAlertmanager) setConfig(cfg alerts.AlertConfigDesc) error {
 	} else if am.cfgs[cfg.User].RawConfig != cfg.RawConfig || hasTemplateChanges {
```

Comment:
It might be unrelated but related - I think we can `userAmConfig.String()` instead of `rawConfig`.

In the case where you have:
```
global:
  smtp_smarthost: 'localhost:25'
  smtp_from: 'youraddress@example.org'
```
and submit:

```
global:
  smtp_from: 'youraddress@example.org'
  smtp_smarthost: 'localhost:25'
```

They're effectively the same but we treat them as different.

Reply:
Doesn't string() hide secrets?

Reply:
Yes, Julien is right. We shouldn't use `String()` because of this.

Reply:
I think we should alert if two different configs like that end up in AM, I'd rather keep the current structure. I don't see why the configs in two different AMs would end up different like that.